src/View/Helper/DocBlockHelper.php

```diff
@@ -0,0 +1,143 @@
+        if ($association->type() === Association::MANY_TO_MANY ||
+            $association->type() === Association::ONE_TO_MANY
+        ) {
+            return $type . '[]';
+        }
+        return $type;
+    }
+
+    /**
+     * Builds a map of entity property names and DocBlock types for use
+     * in generating `@property` hints.
+     *
+     * This method expects a property schema as generated by
+     * `\Bake\Shell\Task\ModelTask::getEntityPropertySchema()`.
+     *
+     * The generated map has the format of
+     *
+     * ```
+     * [
+     *     'property-name' => 'doc-block-type',
+     *     ...
+     * ]
+     * ```
+     *
+     * @see \Bake\Shell\Task\ModelTask::getEntityPropertySchema
+     *
+     * @param array $propertySchema The property schema to use for generating the type map.
+     * @return array The property DocType map.
+     */
+    public function buildEntityPropertyHintTypeMap(array $propertySchema)
+    {
+        $properties = [];
+        foreach ($propertySchema as $property => $info) {
+            switch ($info['kind']) {
+                case 'column':
+                    $properties[$property] = $this->columnTypeToHintType($info['type']);
+                    break;
+
+                case 'association':
+                    $type = $this->associatedEntityTypeToHintType($info['type'], $info['association']);
+                    if ($info['association']->type() === Association::MANY_TO_ONE) {
+                        $properties = $this->_insertAfter(
+                            $properties,
+                            $info['association']->foreignKey(),
+                            [$property => $type]
+                        );
+                    } else {
+                        $properties[$property] = $type;
+                    }
+                    break;
+            }
+        }
+        return $properties;
+    }
+
+    /**
+     * Converts a column type to its DocBlock type counterpart.
+     *
+     * This method only supports the default CakePHP column types,
+     * custom column/database types will be ignored.
+     *
+     * @see \Cake\Database\Type
+     *
+     * @param string $type The column type.
+     * @return null|string The DocBlock type, or `null` for unsupported column types.
+     */
+    public function columnTypeToHintType($type)
+    {
+        switch ($type) {
+            case 'string':
+            case 'text':
+            case 'uuid':
+                return 'string';
+
+            case 'integer':
+            case 'biginteger':
+                return 'int';
+
+            case 'float':
+            case 'decimal':
+                return 'float';
+
+            case 'boolean':
+                return 'bool';
+
+            case 'binary':
+                return 'string|resource';
+
+            case 'date':
+            case 'datetime':
+            case 'time':
+            case 'timestamp':
+                return 'string|int|\DateTime|\Cake\I18n\Time';
```

Comment:
@josegonzalez  But these are all accepted types for date/time based columns. Are there plans for restricting them, or what exactly do you mean by "comfortable"?